solidity/dashboard/src/forms/common-validators.js

```diff
@@ -22,25 +23,43 @@ export const validateEthAddress = (address, required = true) => {
   }
 }
 
+/**
+ * Checks if the value is a valid decimal number.
+ * To be a valid decimal number the value must fulfill three conditions:
+ * - must have a dot (but not at the beginning)
+ * - must have numbers only when we don't consider dots
+ * - must have exactly one occurrence of a dot
+ * @param {string} value given value
+ * @return {boolean}
+ */
+const isValidDecimalNumber = (value) => {
+  if (value.indexOf(".") <= 0) return false
+  const valueWithoutDots = value.replace(".", "")
+  return (
+    REGEXP_ONLY_NUMBERS.test(valueWithoutDots) &&
+    value.match(REGEX_DOT || []).length === 1
+  )
+}
+
 export const validateAmountInRange = (
   value,
   maxValue,
   minValue = 0,
   token = KEEP
 ) => {
-  const formatedValue = value
-    ? web3Utils
-        .toBN(value)
-        .mul(web3Utils.toBN(10).pow(web3Utils.toBN(18)))
-        .toString()
-    : 0
-  const validateValueInBN = web3Utils.toBN(formatedValue)
+  let formattedValue = value
+  try {
+    formattedValue = value ? web3Utils.toWei(value) : 0
```

Comment:
I would suggest using `const formatedValue = token.fromTokenUnit(value)` because not every token has a precision of 18 decimal places. Thoughts? Please see https://github.com/keep-network/keep-core/blob/2a0d1859ec814d3a77394d529abf0214c59027d0/solidity/dashboard/src/forms/common-validators.js#L33-L56 and let me know if this makes sense to you.

Reply:
8ce176f3cc8cd0e842499b0c2d5f7a601d1a3c08